uber/models.py

```diff
@@ -340,8 +340,13 @@ def get_tracking_by_instance(self, instance, action, last_only=True):
         return query.first() if last_only else query.all()
 
     def apply(self, params, *, bools=(), checkgroups=(), restricted=True, ignore_csrf=True):
+        """
+        Args:
+            restricted (bool): if true, restrict any changes only to fields which we allow attendees to set on their own
+                if false, allow changes to any fields.
+        """
         bools = self.regform_bools if restricted else bools
-        checkgroups = self.regform_checkgroups if restricted else bools
+        checkgroups = self.regform_checkgroups if restricted else checkgroups
         for column in self.__table__.columns:
```

Comment:
D'oh!  I'm embarrassed because not only was this my own bug (which is not embarrassing in an of itself), but because this looks like a clear case of me copy/pasting a line of code and then not sufficiently editing it afterwards.  Rookie mistake!

Great catch, awesome that you even wrote unit tests for it!